apps/core/components/ItineraryCard/TripSector.js

```diff
@@ -3,62 +3,52 @@
 import * as React from 'react';
 import { View } from 'react-native';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
-import { Text, StyleSheet, CarrierLogo } from '@kiwicom/universal-components';
+import { Text, StyleSheet } from '@kiwicom/universal-components';
+import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
 
+import type { TripSector as TripSectorType } from './__generated__/TripSector.graphql';
 import TimelineArrow from './TimelineArrow';
+import Transporters from './Transporters';
+import { getFormattedDate, getDuration, dateFormat } from './TripSectorHelpers';
 
 type Props = {|
-  +arrival: string,
-  +arrivalTime: string,
-  +carrier: $ElementType<
-    $PropertyType<React.ElementProps<typeof CarrierLogo>, 'carriers'>,
-    number,
-  >,
-  +tripDate: string,
-  +departure: string,
-  +departureTime: string,
-  +duration: string,
+  +data: ?TripSectorType,
 |};
 
-export default function TripSegment({
-  arrival,
-  arrivalTime,
-  carrier,
-  departure,
-  departureTime,
-  duration,
-  tripDate,
-}: Props) {
+function TripSector({ data }: Props) {
+  const firstSegment = data?.segments?.[0];
+  const lastSegment =
+    data && data.segments && data.segments[data.segments.length - 1];
```

Comment:
please use optionalChaining together with last from rambda here. 

```
const segments = data?.segments ?? [];
const firstSegment = first(segments);
const lastSegment = last(segments);
```

Reply:
Nice, it's much better. Thanks!